### PR TITLE
Handle invalid genes error by directing use to correct query in query…

### PR DESCRIPTION
**Pull request by alisman**
We get a handful of errors each day where users have outdated links with invalid genes.  This PR directs users back to query page to fix things.

### CODE REVIEW

File: src/shared/components/query/QueryAndDownloadTabs.tsx [hunk 1]
```diff
@@ -14,21 +15,24 @@ const QUERY = 'query';
...
 
-	get store()
-	{
-		return this.props.store;
+	constructor(props:IQueryAndDownloadTabsProps){
+		super();
+		this.store = props.getQueryStore();
 	}
 
+	@observable.ref store: QueryStore;
```
Comment (adamabeshouse) [inline src/shared/components/query/QueryAndDownloadTabs.tsx]: you only want to set this on init?

File: src/pages/resultsView/querySummary/QuerySummary.tsx [hunk 2]
```diff
@@ -141,6 +140,16 @@ export default class QuerySummary extends React.Component<{ routingStore:Extende
         getGAInstance()('send', 'event', 'resultsView', 'query modified');
     }
 
+    @computed get queryForm(){
+        return <div style={{marginTop:10}}>
```
Comment (adamabeshouse) [inline src/pages/resultsView/querySummary/QuerySummary.tsx]: welcome to the dark side 😈

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 3]
```diff
@@ -2051,12 +2053,29 @@ export class ResultsViewPageStore {
...
         onResult:(genes:Gene[])=>{
             this.geneCache.addData(genes);
+        },
+        onError:(err)=>{
+            // throwing this allows sentry to report it
+            throw(err);
         }
     });
 
+    @computed get genesInvalid(){
```
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: should we also catch invalid profiles/studies/overall query, like `QueryStore.submitError`?

Comment (alisman) [reply src/pages/resultsView/ResultsViewPageStore.ts]: lets discuss at standup.  i happen to know from sentry that we often get invalid genes.  not sure if that's true for those other params.

File: src/pages/resultsView/querySummary/QuerySummary.tsx (hunk 2 above)
Comment (alisman) [reply src/pages/resultsView/querySummary/QuerySummary.tsx]: why is this the dark side?  jsx from @computed?  if so, i've long been guilty.  the darkside is jsx from mobxpromise.

File: src/shared/components/query/QueryAndDownloadTabs.tsx (hunk 1 above)
Comment (alisman) [reply src/shared/components/query/QueryAndDownloadTabs.tsx]: the basic idea is that when i show the the query component, i want to create a store for that session/lifecycle of the component.    it cannot be replaced/updated by anything.  and it gets throw out on close.   i could model this above in the page, but this seems cleaner to me.  let me know what you think.

**Pull request merged**